### PR TITLE
circleci 0.0.4579-9c6473a

### PR DESCRIPTION
**Pull request by Ketouem**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/circleci.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+class Circleci < Formula
+  desc "Tool which enables to reproduce CircleCI build locally"
+  homepage "https://circleci.com/docs/2.0/local-jobs/"
+  url "https://circle-downloads.s3.amazonaws.com/releases/build_agent_wrapper/circleci"
```
Comment (DomT4) [inline Formula/circleci.rb]: This doesn't seem like a URL that is going to remain stable over time? You need a versioned URL, not one that will always point to the latest.

File: Formula/circleci.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  version "0.0.4579-9c6473a"
+  sha256 "2db865b97d490914e7fb6a056f94d80b1cc9e3b2ed9c09723b529d673c131d7a"
+
+  bottle :unneeded
+
+  depends_on "docker"
+
+  def install
+    bin.install "circleci"
```
Comment (DomT4) [inline Formula/circleci.rb]: Needs to be built from source if this is using `go`, which I suspect it very well might be.

Comment (DomT4) [reply Formula/circleci.rb]: > You can update your local build tool with circleci update.

Also suspect Homebrew may want a way to disable that, looking at things like `heroku`.

File: Formula/circleci.rb (hunk 1 above)
Comment (Ketouem) [reply Formula/circleci.rb]: This is the one given by CircleCI itself https://circleci.com/docs/2.0/local-jobs/ no version specific URL seems to be available.

File: Formula/circleci.rb (hunk 2 above)
Comment (Ketouem) [reply Formula/circleci.rb]: The program is a shell script that invokes a Docker container, so how can I build it from source ?

Comment (Ketouem) [reply Formula/circleci.rb]: The update process downloads a newer Docker image but does not update the CLI itself

**Comment by Ketouem** [comment]
The build error comes from the `-t` option used with Docker inside the installed script, @DomT4 do you know any way to sed temporarily the file to remove this option ?

File: Formula/circleci.rb (hunk 2 above)
Comment (DomT4) [reply Formula/circleci.rb]: > The program is a shell script that invokes a Docker container, so how can I build it from source ?

Oh interesting, I assumed it was a `go` binary or something, as has become incredibly popular. If it's simply a shell script it's obviously fine to not build from source.

> The update process downloads a newer Docker image for the build agent but does not update the CLI itself.

That should _probably_ be okay then, I assume.

File: Formula/circleci.rb (hunk 1 above)
Comment (DomT4) [reply Formula/circleci.rb]: You'll need to ask them for a versioned URL.

**Comment by DomT4** [comment]
>  do you know any way to sed temporarily the file to remove this option ?

`inreplace` is your friend.

**Comment by ilovezfs** [comment]
Closing this out for now as it doesn't look like upstream does actual releases. But if that changes, we can reopen! Thanks for the PR @Ketouem

File: Formula/circleci.rb (hunk 1 above)
Comment (eddiewebb) [reply Formula/circleci.rb]: I've submitted a (PR that would create versioned URLs([https://github.com/circleci/local-cli/issues/7] for use by a formula, and have it working against a (tap in my own repo)[https://github.com/eddiewebb/homebrew-circleci]

If that PR gets merged I will submit a new PR against core. the tap formula is passing all audits except repo age (< 30 days)

**Pull request closed without merging**